ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7297,8 +7435,19 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
       // Add NOT NULL constraint check
       input = genConstraintsPlan(dest, qb, input);
 
-      // Add sorting/bucketing if needed
-      input = genBucketingSortingDest(dest, input, qb, tableDescriptor, destinationTable, rsCtx);
+      if (destinationTable.isMaterializedView() &&
+          mvRebuildMode == MaterializationRebuildMode.INSERT_OVERWRITE_REBUILD) {
+        // Data organization (DISTRIBUTED, SORTED, CLUSTERED) for materialized view
+        // TODO: We only do this for a full rebuild
```

Comment:
I think a `WARN` may be too aggressive, since there is nothing wrong in the system, it is just a feature that is missing (incremental maintenance of the view does not organize the new data according to the spec). I will add a DEBUG message though.

Reply:
I was thinking about this and there is no need to log this info. This is a missing feature, and there is a `TODO` and a follow-up JIRA (https://issues.apache.org/jira/browse/HIVE-21953).